library/src/blas3/rocblas_trmm.hpp

```diff
@@ -0,0 +1,1453 @@
+                                                               rb,
+                                                               0,
+                                                               &c[ii - 1 + (j - 1) * ldc],
+                                                               0,
+                                                               1,
+                                                               0,
+                                                               1));
+                        }
+                    }
+                    //
+                    //                  C := alpha*A*C + C, general matrix multiply
+                    //                  involving a rectangular block of A.
+                    //
+                    if(ii + isec <= m)
+                    {
+                        PRINT_AND_RETURN_IF_ROCBLAS_ERROR(
+                            (rocblas_gemm_template<false, false>)(handle,
+                                                                  rocblas_operation_none,
+                                                                  rocblas_operation_none,
+                                                                  isec,
+                                                                  n,
+                                                                  m - ii - isec + 1,
+                                                                  alpha,
+                                                                  (const T*)(&a[ii - 1
+                                                                                + (ii + isec - 1)
+                                                                                      * lda]),
+                                                                  0,
+                                                                  lda,
+                                                                  0,
+                                                                  (const T*)(&c[ii + isec - 1]),
+                                                                  0,
+                                                                  ldc,
+                                                                  0,
+                                                                  &one,
+                                                                  &c[ii - 1],
+                                                                  0,
+                                                                  ldc,
+                                                                  0,
+                                                                  1));
+                    }
+                }
+            }
+            else
+            {
+                //
+                //             Form  C := alpha*A'*C. Left, Upper, Transpose.
+                //
+                bool cldc = dcld(ldc);
+                for(int ii = m - ((m - 1) % cb); ii >= 1; ii -= cb)
```

Comment:
It's not important now, but for consistency we should use `rocblas_int` for loop control variables limited by `rocblas_int` variables.

If the loop limit is calculated with multiplication, and thus can exceed 32-bits, at least one operand should be promoted with `size_t(n)` and the product and any loops which iterate over it should be `size_t`.